File: src/main/scala/com/pwootage/oc/js/api/JSComputerApi.scala
package com.pwootage.oc.js.api

import java.util

import com.pwootage.oc.js.{JSEngine, JSUtils, OCSignalHandler}
import li.cil.oc.api.machine.Machine
import li.cil.oc.api.network.Connector

import scala.collection.JavaConversions._

class JSComputerApi(machine: Machine, engine: JSEngine) {
  def address(): String = machine.node().address()

  def tmpAddress(): String = machine.tmpAddress()

  def freeMemory(): Int = engine.maxMemory - engine.usedMemory

  def totalMemory(): Int = engine.maxMemory

  def energy(): Double = machine.node.asInstanceOf[Connector].globalBuffer

  def maxEnergy(): Double = machine.node.asInstanceOf[Connector].globalBufferSize

  def uptime(): Double = machine.upTime()

  //TODO: Handle users eventually
}

File: src/main/java/com/pwootage/oc/js/api/JSComputerApi.kt
package com.pwootage.oc.js.api

import com.pwootage.oc.js.JSEngine
import li.cil.oc.api.machine.Machine
import li.cil.oc.api.network.Connector


class JSComputerApi(val machine: Machine, val engine: JSEngine) {
  fun address(): String = machine.node().address()

  fun tmpAddress(): String = machine.tmpAddress()

  fun freeMemory(): Int = engine.maxMemory - engine.usedMemory

  fun totalMemory(): Int = engine.maxMemory

  fun energy(): Double = (machine.node() as Connector).globalBuffer()

  fun maxEnergy(): Double = (machine.node() as Connector).globalBufferSize()

  fun uptime(): Double = machine.upTime()

  //TODO: Handle users eventually
}
